### ISSUE TITLE
Type Orchard and Sapling note commitment `x`/`u` coordinates instead of a plain `Base` types

### ISSUE THREAD

**Issue by dconnolly** [ZcashFoundation/zebra#3028]
## Scheduling

This task is blocked by ZSA specification and shared Rust crate implementation, which might require changes to the same code.

## Motivation

Currently we have defined the Orchard note commitment x-coordinate as just the 'raw' `pasta_curves::pallas::Base` type that it is, and for Sapling the equivalent u-coordinate is just a `jubjub::Base` type. To transform other types such as `[u8; 32]` into these types, we need to either implement our own owned traits or import the non-std traits they do, pulling in other dependencies into our tree.

### Design

We can align these types with our `Sprout` note commitments and other types in `zebra-chain` as a tuple struct that wraps the coordinate, and implement the std library `From/Into` or other traits on the wrapper type, which will allow us to encapsulate any dependencies within `zebra-chain` instead of forcing consumers of these types (eg: `zebra-consensus`, `zebra-client`) to also import those dependencies. 

```rust
// orchard

pub struct 
NoteCommitmentCoordinate(pallas::Base);

// sapling

pub struct NoteCommitmentCoordinate(jubjub::Fq);

```

Everywhere where this is represented (`ShieldedData`) will need to be updated.

-[ ] `sapling::commitment::NoteCommitmentCoordinate`
-[ ] `orchard::commitment::NoteCommitmentCoordinate`

or something like that
 
### Alternatives

Replace these `zebra_chain::orchard` types with existing types from the `orchard` and `librustzcash` crates.

### Tasks

- [ ] Switch zebra-chain::orchard::NoteCommitment to the one in exernal orchard crate
- [ ] Use ExtractedNoteCommitment in orchard
- [ ] Making changes in ext. orchard to use NoteCommitment in zebra (make API public)
- [x] Talk to QEDIT about these changes

- [ ] Switch zebra-chain::sapling::NoteCommitment to the one in external sapling-crypto crate
- [ ] Use ExtractedNoteCommitment in sapling

**Comment by dconnolly** [comment]
This is still a valid issue 👍

**Comment by teor2345** [comment]
We might also want to consider re-using existing types from the `orchard` and `librustzcash` crates, particularly after they are updated for ZSAs.

**Comment by mpguerra** [comment]
related to #8155

**Comment by natalieesk** [comment]
Closing as superseded by migrating to upstream orchard/sapling crate types (#8155).
